File: backend/app/services/product_service.py
import logging
from typing import List, Optional

logger = logging.getLogger(__name__)

class ProductService:
    """Service for handling product-related operations"""
    
    async def search_products(self, query: str, platforms: List[str] = None) -> List[dict]:
        """Search for products across platforms"""
        logger.info(f"Searching for products: {query}")
        # Implementation will be added
        return []
    
    async def get_product_by_id(self, product_id: str) -> Optional[dict]:
        """Get product details by ID"""
        logger.info(f"Fetching product: {product_id}")
        # Implementation will be added
        return None
    
    async def get_price_history(self, product_id: str, days: int = 30) -> List[dict]:
        """Get price history for a product"""
        logger.info(f"Fetching price history for: {product_id}")
        # Implementation will be added
        return []

class ComparisonService:
    """Service for comparing prices across platforms"""
    
    async def compare_prices(self, product_id: str) -> dict:
        """Compare product prices across all platforms"""
        logger.info(f"Comparing prices for: {product_id}")
        # Implementation will be added
        return {}
    
    async def find_best_deal(self, product_id: str) -> dict:
        """Find the best deal for a product"""
        logger.info(f"Finding best deal for: {product_id}")
        # Implementation will be added
        return {}

class AlertService:
    """Service for handling price alerts"""
    
    async def create_alert(self, product_id: str, target_price: float, email: Optional[str] = None) -> dict:
        """Create a price alert"""
        logger.info(f"Creating alert for product: {product_id}")
        # Implementation will be added
        return {}
    
    async def check_alerts(self) -> List[dict]:
        """Check and trigger alerts"""
        logger.info("Checking price alerts")
        # Implementation will be added
        return []

File: frontend/src/utils/productService.js
import apiClient from './apiClient';

export const searchProducts = async (query, platforms = []) => {
  try {
    const response = await apiClient.get('/products/search', {
      params: { q: query, platforms: platforms.join(',') }
    });
    return response.data;
  } catch (error) {
    console.error('Search error:', error);
    throw error;
  }
};

export const getProductDetails = async (productId) => {
  try {
    const response = await apiClient.get(`/products/${productId}`);
    return response.data;
  } catch (error) {
    console.error('Fetch product error:', error);
    throw error;
  }
};

export const getPriceHistory = async (productId) => {
  try {
    const response = await apiClient.get(`/products/${productId}/prices`);
    return response.data;
  } catch (error) {
    console.error('Fetch price history error:', error);
    throw error;
  }
};

export const comparePrices = async (productId) => {
  try {
    const response = await apiClient.get(`/compare/${productId}`);
    return response.data;
  } catch (error) {
    console.error('Compare prices error:', error);
    throw error;
  }
};

export const createAlert = async (productId, targetPrice) => {
  try {
    const response = await apiClient.post('/alerts', {
      product_id: productId,
      target_price: targetPrice
    });
    return response.data;
  } catch (error) {
    console.error('Create alert error:', error);
    throw error;
  }
};

export const getAlerts = async () => {
  try {
    const response = await apiClient.get('/alerts');
    return response.data;
  } catch (error) {
    console.error('Fetch alerts error:', error);
    throw error;
  }
};
